update Banner, Field & ScrollBox

storywithoutend opened the pull request:
UPDATE
Since last review, here are the following changes
* Textarea - Made the action button customizable so it can handle something other than "clear". Also fixed a bug where the padding on the right is now there for both clearable and custom action events.
* Input - Added new property that allows for adjusting the size of the icon used. Also added a hover state for the action/clear button.
* Avatar - This is from another PR and I added it in, but when the src of the avatar image is set to falsy, we need a condition to show the placeholder again.
-------------------------
* Change Banner message type to ReactNode... I think we should consider removing message and turning it to children.
* Make Field labels not overflow.
* Fix horizontal scroll tab on ScrollBox

TateB reviewed (approved):
change version number back before merge but lgtm

TateB reviewed (changes requested):
- hover state for action button in textarea
- pointer cursor for action button in textarea/input

storywithoutend commented:
> * hover state for action button in textarea
> * pointer cursor for action button in textarea/input

Add hover states
removed requirement of href on Banner props
Added padding on right hand side of Banner when there is action button
